api/rs/build/lib.rs

```diff
@@ -104,6 +104,30 @@ impl CompilerConfiguration {
 
     /// Create a new configuration that sets the library paths used for looking up
     /// `@library` imports to the specified map of paths.
+    ///
+    /// Each library path can either be a path to a `.slint` file or a directory.
+    /// If it's a file, the library is imported by its name prefixed by `@` (e.g.
+    /// `@example`). The specified file is the only entry-point for the library
+    /// and other files from the library won't be accessible from the outside.
+    /// If it's a directory, a specific file in that directory must be specified
+    /// when importing the library (e.g. `@example/widgets.slint`). This allows
+    /// exposing multiple entry-points for a single library.
+    ///
+    /// Compile `ui/main.slint` and specify an "example" library path:
+    /// ```rust,no_run
+    /// let manifest_dir = std::path::PathBuf::from(std::env::var_os("CARGO_MANIFEST_DIR").unwrap());
+    /// let library_paths = std::collections::HashMap::from([(
+    ///     "example".to_string(),
+    ///     manifest_dir.join("third_party/example/ui/lib.slint"),
+    /// )]);
```

Comment:
This is not really what i had in mind.
I was thinking that the code that use CARGO_MANIFEST_DIR would be in a build-dependency (eg, coop_widget uploaded to crates.io)
but that work too.

Reply:
I have somewhat mixed feelings about teaching users to wire up Cargo dependencies manually if the end goal is to make it automatic once the issues with compiling libraries have been sorted out. :)

Reply:
Ok, i understand.  Just leave it like this then.